marin/processing/tokenize/tokenize.py

```diff
@@ -112,6 +113,32 @@ def __post_init__(self):
         if isinstance(self.validation_paths, Sequence):
             assert "/" not in self.validation_paths, "don't use the entire fs for validation paths!"
 
+        _validate_train_urls(self.train_paths)
+
+
+def _validate_train_urls(train_paths: list[str | InputName]):
+    """
+    Validates the training data URLs or InputName attributes to ensure they do not contain forbidden patterns.
+    Raises a ValueError if a forbidden pattern is found.
+    """
+    for item in train_paths:
+        url_or_name_to_check = None
+        if isinstance(item, str):
+            url_or_name_to_check = item
+        elif hasattr(item, "name") and isinstance(item.name, str):
+            url_or_name_to_check = item.name
+
+        if url_or_name_to_check:
+            # \b doesn't work because of underscores
+            if re.search(r"[^a-zA-Z]test[^a-zA-Z]", url_or_name_to_check) or re.search(
+                r"validation", url_or_name_to_check
+            ):
+                raise ValueError(
```

Comment:
Should we print a warning?  what if we wanted to actually train on test (e.g., to test the effects of training on test)?  It seems like a hard prohibition is a bit strict?